src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
```

Comment:
IMO first extract the SwingWorker + progress bar to be generic. `SwingComponents.java` is a default place for swing component factory methods.

Then you could do something like:
- have Screenshot be defined in a class that implements Runnable
- you could then write something like `SwingComponents.runWithProgressBar("message", SaveScreenshot::new)`